specification/devcenter/data-plane/Microsoft.DevCenter/preview/2022-11-11-preview/devbox.json

```diff
@@ -0,0 +1,1474 @@
+            "schema": {
+              "$ref": "devcenter.json#/definitions/CloudError"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "The error code for specific error that occurred.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DevBoxes_CreateDevBox": {
+            "$ref": "./examples/DevBoxes_Create.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Dev Boxes"
+        ],
+        "parameters": [
+          {
+            "$ref": "devcenter.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "devcenter.json#/parameters/ProjectNameParameter"
+          },
+          {
+            "$ref": "devcenter.json#/parameters/UserIdParameter"
+          },
+          {
+            "$ref": "#/parameters/DevBoxNameParameter"
+          }
+        ],
+        "description": "Deletes a Dev Box.",
+        "operationId": "DevBoxes_DeleteDevBox",
+        "x-ms-long-running-operation": true,
+        "responses": {
+          "202": {
+            "description": "The request was accepted",
+            "headers": {
+              "Operation-Location": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "Resource does not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "devcenter.json#/definitions/CloudError"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "The error code for specific error that occurred.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DevBoxes_DeleteDevBox": {
+            "$ref": "./examples/DevBoxes_Delete.json"
+          }
+        }
+      }
+    },
+    "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}:start": {
+      "post": {
+        "tags": [
+          "Dev Boxes"
+        ],
+        "parameters": [
+          {
+            "$ref": "devcenter.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "devcenter.json#/parameters/ProjectNameParameter"
+          },
+          {
+            "$ref": "devcenter.json#/parameters/UserIdParameter"
+          },
+          {
+            "$ref": "#/parameters/DevBoxNameParameter"
+          }
+        ],
+        "description": "Starts a Dev Box",
+        "operationId": "DevBoxes_StartDevBox",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "responses": {
+          "202": {
```

Comment:
Presumably this is returned when the LRO is started.  What's returned on terminal success?

Reply:
The previous version was never a valid response. We do final state via location, so the return on terminal success is just a succeeded/failed operation status. The 200 response was present in our spec due to a miscommunication with the API board; we never actually returned a DevBox schema on terminal state from the monitor URL. Our understanding is that we don't need anything except the 202 here if the terminal response is an operation status, but let me know if that's not accurate!

Reply:
If we do still need to include something special just for codegen -- not sure what the best practice is considered, but personally I would prefer to do that via directives in the autorest configs rather than have incorrect info in the spec

Reply:
Does this mean that the "final GET" on the URL in the `Location` header returns a 204 on terminal success?

Reply:
GET on the status monitor here will always return 200 response code (assuming valid operation ID/permissions etc.). We're following the Operation-Location guidance from the Microsoft REST guidelines for that. The status property will indicate state, so when the operation reaches a terminal state, the response on the final GET would be 200 + operation body that looks like this:
{
        "id": "/projects/{projectName}/operationStatuses/{operationId}",
        "name": "{operationId}",
        "status": "Succeeded",
        "startTime": "2022-09-28T22:55:29.0911599\u002B00:00",
        "endTime": "2022-09-28T23:04:04.4337491\u002B00:00"
}

Reply:
The `final-state-via` setting, which has a confusing name in hindsight, describes which URL we should GET to fetch the schema after the LRO has reached terminal success, not the state (i.e. `status` in the above example) of the operation.  This "final GET" happens after the LRO's status has reached terminal success.

For this operation, since no schema is returned on terminal success, there is no need to perform a final GET afterward.  So IMO this part can be removed.

This is how it works for control-plane, and I believe we have the same semantics for data-plane.  @mikekistler do you have anything to add?

Reply:
Agreed with @jhendrixMSFT that you can just remove "x-ms-long-running-operation-options".